Content.Shared/_Harmony/ItemCurse/SharedItemCurseSystem.cs

```diff
@@ -0,0 +1,211 @@
+        _throwing.TryThrow(ent, _random.NextVector2(), baseThrowSpeed: comp.FlingStrength);
+        CreateLightning(ent, comp);
+    }
+
+    private void OnCurseMarkerShutdown(Entity<CurseMarkerComponent> ent, ref ComponentShutdown args)
+    {
+        TryUnmarkItem(ent);
+    }
+
+    private void TryMarkItem(Entity<ItemCurseComponent> ent, EntityUid item)
+    {
+        if (_actions.GetAction(ent.Owner) is not { } action)
+            return;
+
+        if (action.Comp.AttachedEntity is not { } user)
+            return;
+
+        ent.Comp.MarkedEntity = item;
+        Dirty(ent);
+
+        var marker = AddComp<CurseMarkerComponent>(item);
+        marker.MarkedByAction = ent;
+        Dirty(item, marker);
+
+        UpdateActionAppearance((action, action, ent));
+    }
+
+    private void TryUnmarkItem(Entity<CurseMarkerComponent> item)
+    {
+        if (!TryComp<CurseMarkerComponent>(item, out var marker))
+            return;
+
+        if (_actions.GetAction(marker.MarkedByAction) is not { } action)
+            return;
+
+        if (TryComp<ItemCurseComponent>(action, out var itemCurse))
+        {
+            // The following comment was from the recall code I copied. I'll just leave it here because it's probably important. TL;DR this code doesn't work yet.
+            //
+            // For some reason client thinks the station grid owns the action on client and this doesn't work. It doesn't work in PopupEntity(mispredicts) and PopupPredicted either(doesnt show).
+            // I don't have the heart to move this code to server because of this small thing.
+            // This line will only do something once that is fixed.
+            if (action.Comp.AttachedEntity is { } user)
+            {
+                _popups.PopupClient(Loc.GetString("item-recall-item-unmark", ("item", item)), user, user, PopupType.MediumCaution);
+            }
+
+            itemCurse.MarkedEntity = null;
+            UpdateActionAppearance((action, action, itemCurse));
+            Dirty(action, itemCurse);
+        }
+
+        RemCompDeferred<CurseMarkerComponent>(item);
+    }
+
+    private void UpdateActionAppearance(Entity<ActionComponent, ItemCurseComponent> action)
+    {
+        if (action.Comp2.MarkedEntity is { } marked)
+        {
+            if (action.Comp2.WhileMarkedName is { } name)
+                _metaData.SetEntityName(action, Loc.GetString(name, ("item", marked)));
+
+            if (action.Comp2.WhileMarkedDescription is { } desc)
+                _metaData.SetEntityDescription(action, Loc.GetString(desc, ("item", marked)));
+
+            _actions.SetEntityIcon((action, action), marked);
+        }
+        else
+        {
+            if (action.Comp2.InitialName is { } name)
+                _metaData.SetEntityName(action, name);
+            if (action.Comp2.InitialDescription is { } desc)
+                _metaData.SetEntityDescription(action, desc);
+            _actions.SetEntityIcon((action, action), null);
```

Comment:
Add spaces like above so it's easier to read
```suggestion
            if (action.Comp2.InitialName is { } name)
                _metaData.SetEntityName(action, name);

            if (action.Comp2.InitialDescription is { } desc)
                _metaData.SetEntityDescription(action, desc);

            _actions.SetEntityIcon((action, action), null);
```